src/Field.tsx

```diff
@@ -84,110 +66,108 @@ export interface FieldConfig {
 
 export type FieldAttributes<T> = GenericFieldHTMLAttributes & FieldConfig & T;
 
-/**
- * Custom Field component for quickly hooking into Formik
- * context and wiring up forms.
- */
-class FieldInner<Values = {}, Props = {}> extends React.Component<
-  FieldAttributes<Props> & { formik: FormikContext<Values> },
-  {}
-> {
-  constructor(
-    props: FieldAttributes<Props> & { formik: FormikContext<Values> }
-  ) {
-    super(props);
-    const { render, children, component } = props;
-    warning(
-      !(component && render),
-      'You should not use <Field component> and <Field render> in the same <Field> component; <Field component> will be ignored'
+export function useField<Val = any>(name: string, type?: string) {
+  const formik = useFormikContext();
+
+  invariant(
+    formik,
+    'useField() / <Field /> must be used underneath a <Formik> component or withFormik() higher order component'
+  );
+
+  return formik.getFieldProps<Val>(name, type);
```

Comment:
with this change one can't use `useField` to get `form` bag where the imperative methods such as `setXXXX` available. Those methods needed to use with third-party libs such as `react-select`.  Would you recommend using `Field` in that case?

Reply:
See the 3 comments right above this :)